packages/core/src/Cardano/types/Address.ts

```diff
@@ -27,3 +30,23 @@ export const Address = (value: string): Address => {
   }
   throw new InvalidStringError(`Invalid address: ${value}`);
 };
+
+/**
+ * Transform hex-encoded bytes address into `Cardano.Address`,
+ * base58 in case of Byron era or bech32 in case of Shelley era or newer.
+ *
+ * @param value hex-encoded bytes address.
+ * @throws {InvalidStringError} if value is invalid
+ */
+Address.fromHex = (value: string): Address =>
```

Comment:
Maybe we could update the existing `Address()` function to use this. It should currently throw given hex-encoded address, so we could call this in the `catch` block

Reply:
Nice, that would be quite elegant. Taking this approach would mean a single commit @mirceahasegan, as it is just a fix for the current function

Reply:
Updated in [2bf0f29](https://github.com/input-output-hk/cardano-js-sdk/pull/564/commits/2bf0f29a886dcf145317864d6dda84d7e9e33c0a)